apps/explorer/src/components/Activity/TransactionsActivityTable.tsx

```diff
@@ -0,0 +1,133 @@
+    initialLimit = DEFAULT_TRANSACTIONS_LIMIT,
+}: Props) {
+    const [currentPage, setCurrentPage] = useState(0);
+    const [limit, setLimit] = useState(initialLimit);
+    const rpc = useRpcClient();
+    const { data, isLoading, isFetching, fetchNextPage, hasNextPage } =
+        useGetTransactionBlocks(undefined, limit);
+
+    const countQuery = useQuery(
+        ['transactions', 'count'],
+        () => rpc.getTotalTransactionBlocks(),
+        { cacheTime: 24 * 60 * 60 * 1000, staleTime: Infinity, retry: false }
+    );
+
+    const cardData =
+        data && Boolean(data.pages[currentPage])
+            ? genTableDataFromTxData(data.pages[currentPage].data)
+            : undefined;
+
+    return (
+        <div data-testid="tx">
+            <div className="flex flex-col space-y-5 text-left xl:pr-10">
+                {isLoading || isFetching || !cardData ? (
+                    <PlaceholderTable
+                        rowCount={limit}
+                        rowHeight="16px"
+                        colHeadings={[
+                            'Digest',
+                            'Sender',
+                            'Txns',
+                            'Gas',
+                            'Time',
+                        ]}
+                        colWidths={['30%', '30%', '10%', '20%', '10%']}
+                    />
+                ) : (
+                    <div>
+                        <TableCard
+                            data={cardData.data}
+                            columns={cardData.columns}
+                        />
+                    </div>
+                )}
+
+                <div className="flex justify-between">
+                    {(hasNextPage || data?.pages.length) &&
+                    !disablePagination ? (
+                        <Pagination
+                            onNext={() => {
+                                if (isLoading || isFetching) {
+                                    return;
+                                }
+
+                                // Make sure we are at the end before fetching another page
+                                if (
+                                    data &&
+                                    currentPage === data?.pages.length - 1
+                                ) {
+                                    fetchNextPage();
+                                }
+                                setCurrentPage(currentPage + 1);
+                            }}
+                            hasNext={
+                                Boolean(hasNextPage) &&
+                                Boolean(data?.pages[currentPage])
+                            }
+                            hasPrev={currentPage !== 0}
+                            onPrev={() => setCurrentPage(currentPage - 1)}
+                            onFirst={() => setCurrentPage(0)}
+                        />
+                    ) : (
+                        disablePagination && (
```

Comment:
`disablePagination` check here is redundant since it's always going to be true due to line 75